src/commands/force/user/permset/assign.ts

```diff
@@ -43,29 +42,27 @@ export class UserPermsetAssignCommand extends SfdxCommand {
       description: messages.getMessage('flags.onBehalfOf'),
     }),
   };
-  private usernames: string[];
   private readonly successes: SuccessMsg[] = [];
   private readonly failures: FailureMsg[] = [];
 
   public async run(): Promise<Result> {
     try {
-      this.usernames = (this.flags.onbehalfof as string[]) ?? [this.org.getUsername()];
+      const aliasOrUsernames = (this.flags.onbehalfof as string[]) ?? [this.org.getUsername()];
 
       const connection: Connection = this.org.getConnection();
       const org = await Org.create({ connection });
 
-      for (const username of this.usernames) {
-        // Convert any aliases to usernames
-        const aliasOrUsername = (await Aliases.fetch(username)) || username;
+      for (const aliasOrUsername of aliasOrUsernames) {
+        // Attempt to convert any aliases to usernames.  Not found alias will be **assumed** to be a username
+        const username = (await Aliases.fetch(aliasOrUsername)) || aliasOrUsername;
         const user: User = await User.create({ org });
         // get userId of whomever the permset will be assigned to via query to avoid AuthInfo if remote user
-        const queryResult: QueryResult<{ Id: string }> = await connection.query(
+        const queryResult = await connection.singleRecordQuery<{ Id: string }>(
```

Comment:
@mshanemc I know this is how the code was written, but can it be optimized to run a single query for all of the usernames present?

Reply:
that code was recently updated in a different PR.  It only got merged here because the nut was failing (correctly! and would have caught our bug!)

I'm not going to fix it as part of the nut.   There's not a ton to be gained because most scratch orgs only have a few users of capacity AND the core.permsetAssign is 1:1.  It'd be more performant to Promise.all the whole loop so each user can 1) get the id and 2) assign the permset in parallel.